bats/tests/k8s/up-downgrade-k8s.bats

```diff
@@ -0,0 +1,132 @@
+# Test case 8 & 13
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start rancher desktop' {
+    start_kubernetes
+    wait_for_apiserver
+    # the docker context "rancher-desktop" may not have been written
+    # even though the apiserver is already running
+    wait_for_container_engine
+}
+
+@test 'deploy nginx - always restart' {
+    ctrctl pull nginx
+    run ctrctl run -d -p 8585:80 --restart=always --name nginx-restart nginx
+    assert_success
+}
+
+@test 'deploy nginx - no restart' {
+    run ctrctl run -d -p 8686:80 --restart=no --name nginx-no-restart nginx
+    assert_success
+}
+
+@test 'deploy busybox' {
+    run kubectl create deploy busybox --image=busybox --replicas=2 -- /bin/sh -c "sleep inf"
+    assert_success
+}
+
+verify_nginx() {
+    for port in 8585 8686; do
+        run curl "http://localhost:$port"
+        assert_success
+        assert_output --partial "Welcome to nginx!"
+    done
+}
+
+@test 'verify nginx before upgrade' {
+    try verify_nginx
+}
+
+verify_busybox() {
+    run kubectl get pods --selector="app=busybox" -o jsonpath='{.items[*].status.phase}'
+    assert_output --partial "Running Running"
+}
+
+@test 'verify busybox before upgrade' {
+    try verify_busybox
+}
+
+verify_images() {
+    if using_docker; then
+        run ctrctl images
+        assert_output --partial "nginx" "busybox"
+    else
+        run ctrctl images --format json
+        assert_output --partial '"Repository":"nginx'
+        run ctrctl images --namespace k8s.io
+        assert_output --partial "busybox"
+    fi
+}
+@test 'verify images before upgrade' {
+    verify_images
+}
+
+@test 'upgrade kubernetes' {
+    rdctl set --kubernetes-version "$RD_KUBERNETES_VERSION"
```

Comment:
We still silently support the old option names, but let's move to the new names in the tests:

```suggestion
    rdctl set --kubernetes.version "$RD_KUBERNETES_VERSION"
```